src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+            //   2. If there are no instances set to process messages, Offset will be null and LastEnqueuedSequenceNumber will be >= 0
+            if (partitionLeaseInfo.Offset == null && partitionInfo.LastEnqueuedSequenceNumber >= 0)
+            {
+                return (partitionInfo.LastEnqueuedSequenceNumber + 1);
+            }
+
+            if (partitionInfo.LastEnqueuedSequenceNumber > partitionLeaseInfoSequenceNumber)
+            {
+                return (partitionInfo.LastEnqueuedSequenceNumber - partitionLeaseInfoSequenceNumber);
+            }
+
+            // Partition is a circular buffer, so it is possible that
+            // LastEnqueuedSequenceNumber < SequenceNumber
+            long count = 0;
+            unchecked
+            {
+                count = (long.MaxValue - partitionInfo.LastEnqueuedSequenceNumber) + partitionLeaseInfoSequenceNumber;
+            }
+
+            // It's possible for checkpointing to be ahead of the partition's LastEnqueuedSequenceNumber,
+            // especially if checkpointing is happening often and load is very low.
+            // If count is negative, we need to know that this read is invalid, so return 0.
+            // e.g., (9223372036854775807 - 10) + 11 = -9223372036854775808
+            return (count < 0) ? 0 : count;
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            return GetScaleStatusCore(context.WorkerCount, context.Metrics?.Cast<EventHubTriggerMetrics>().ToArray());
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext<EventHubTriggerMetrics> context)
+        {
+            return GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
+        }
+
+        private ScaleStatus GetScaleStatusCore(int workerCount, EventHubTriggerMetrics[] metrics)
+        {
+            ScaleStatus status = new ScaleStatus
+            {
+                Vote = ScaleVote.None
+            };
+
+            const int NumberOfSamplesToConsider = 5;
+
+            // Unable to determine the correct vote with no metrics.
+            if (metrics == null || metrics.Length == 0)
+            {
+                return status;
+            }
+
+            // We shouldn't assign more workers than there are partitions
+            // This check is first, because it is independent of load or number of samples.
+            int partitionCount = metrics.Last().PartitionCount;
+            if (partitionCount > 0 && partitionCount < workerCount)
+            {
+                status.Vote = ScaleVote.ScaleIn;
+                _logger.LogInformation($"WorkerCount ({workerCount}) > PartitionCount ({partitionCount}).");
+                _logger.LogInformation($"Number of instances ({workerCount}) is too high relative to number " +
+                                       $"of partitions for EventHub entity ({_eventHubName}, {partitionCount}).");

```

Comment:
Is this last line of the error message supposed to be ({_eventHubName}, {consumerGroup})?

Reply:
This is actually consistent with the logs for the other extensions. Looking back to the first extension I did, I think I was trying to mimic [this log]() but looking at it now, that Queue log makes a lot more sense than this one. I agree it could be clearer that I'm referencing partitioncount. I can change this now and make an issue to change the rest of them